### PR TITLE
Orca: update doc for pytorch estimator backend.

### PR DESCRIPTION
**Pull request by lalalapotter**
## Description

This PR is to update document for the change of PyTorch Estimator backend default value.

### CODE REVIEW

File: docs/readthedocs/source/doc/Orca/Overview/distributed-training-inference.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ View the related [Python API doc](https://bigdl.readthedocs.io/en/latest/doc/Pyt
 
 **Using *BigDL* backend**
 
-Users may create a PyTorch `Estimator` using the *BigDL* backend (currently default for PyTorch) as follows:
+Users may create a PyTorch `Estimator` using the *Spark* backend (currently default for PyTorch) as follows:
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/Overview/distributed-training-inference.md:160]: also need to change the code? e.g. should use model_creator?

Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/Overview/distributed-training-inference.md:160]: Fixed.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 2]
```diff
@@ -1,43 +1,34 @@
...
 pip install torch==1.7.1 torchvision==0.8.2
-pip install six cloudpickle
-pip install jep==3.9.0
 ```
 
 ### Step 1: Init Orca Context
 ```python
 from bigdl.orca import init_orca_context, stop_orca_context
 
-cluster_mode = "local"
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:29]: keep this?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 3]
```diff
@@ -1,43 +1,34 @@
...
 
 ---
 
-![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb)
+![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb)
 
 ---
 
-**In this guide we will describe how to scale out _PyTorch_ programs using Orca in 4 simple steps.**
+**In this guide we will describe how to scale out _PyTorch_ programs using the `torch.distributed` package in Orca.**
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:9]: why this change?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 4]
```diff
@@ -1,43 +1,34 @@
-# PyTorch Quickstart
+# Use `torch.distributed` in Orca
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:1]: why this change?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 5]
```diff
@@ -71,43 +62,53 @@ class LeNet(nn.Module):
         x = self.fc2(x)
         return F.log_softmax(x, dim=1)
 
-model = LeNet()
-model.train()
 criterion = nn.NLLLoss()
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:65]: put this together with model_creator?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 6]
```diff
@@ -71,43 +62,53 @@ class LeNet(nn.Module):
...
+    return model
+
+def optim_creator(model, config):
+    return torch.optim.Adam(model.parameters(), lr=0.001)
 ```
 
 ### Step 3: Define Train Dataset
 
-You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html). 
+You can define the dataset using a *Data Creator Function* that returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](../Overview/data-parallel-processing).
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: Add Spark DataFrame?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 7]
```diff
@@ -71,43 +62,53 @@ class LeNet(nn.Module):
...
 
 ```python
 import torch
 from torchvision import datasets, transforms
 
 torch.manual_seed(0)
-dir='./'
-
-batch_size=64
-test_batch_size=64
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:92]: batch 64 or 320?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 8]
```diff
@@ -71,43 +62,53 @@ class LeNet(nn.Module):
...
-    batch_size=batch_size, shuffle=True)
-test_loader = torch.utils.data.DataLoader(
-    datasets.MNIST(dir, train=False,
-                   transform=transforms.Compose([
-                       transforms.ToTensor(),
-                       transforms.Normalize((0.1307,), (0.3081,))
-                   ])),
-    batch_size=test_batch_size, shuffle=False)
+batch_size = 320
+test_batch_size = 320
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:88]: remove this and just use batch_size?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 9]
```diff
@@ -116,18 +117,14 @@ First, Create an Estimator
 from bigdl.orca.learn.pytorch import Estimator 
 from bigdl.orca.learn.metrics import Accuracy
 
-est = Estimator.from_torch(model=model, optimizer=adam, loss=criterion, metrics=[Accuracy()], backend="bigdl")
+est = Estimator.from_torch(model=model_creator, optimizer=optim_creator, loss=criterion, metrics=[Accuracy()], backend="bigdl")
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:120]: backend is wrong

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 10]
```diff
@@ -140,10 +137,10 @@ Save the Estimator states (including model and optimizer) to the provided model
 est.save("mnist_model")
 ```
 
-Load the Estimator states (model and possibly with optimizer) from the provided model path.
+Load the Estimator states (model and possibly with optimizer) from provided model path.
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:140]: revert this change?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 6 above)
Comment (hkvision) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: SparkXShards -> XShards

File: python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb [hunk 11]
```diff
@@ -0,0 +1,502 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "view-in-github"
+      },
+      "source": [
+        "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_distributed_lenet_mnist.ipynb)"
```
Comment (hkvision) [inline python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb:10]: is this link correct?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 6 above)
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: Need to align the contents with the notebook.
Notebook it write like this `You can define the dataset using a Data Creator Function that has two parameters config and batch_size and returns a PyTorch DataLoader.`

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 8 above)
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:88]: remove this and just use batch_size?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 12]
```diff
@@ -71,43 +62,53 @@ class LeNet(nn.Module):
...
-test_loader = torch.utils.data.DataLoader(
-    datasets.MNIST(dir, train=False,
-                   transform=transforms.Compose([
-                       transforms.ToTensor(),
-                       transforms.Normalize((0.1307,), (0.3081,))
-                   ])),
-    batch_size=test_batch_size, shuffle=False)
+batch_size = 320
+test_batch_size = 320
+dir = './dataset'
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:89]: need to align with the notebook, notebook uses this: dir = '/tmp/dataset'

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 4 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:1]: No need, fixed.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 3 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:9]: No need, fixed.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 2 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:29]: fixed.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 7 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:92]: tested on 320，64 may be work as well.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 13]
```diff
@@ -2,26 +2,24 @@
...
 ---
 
-![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb)
+![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb)
 
 ---
 
-**In this guide we will describe how to scale out _PyTorch_ programs using Orca in 4 simple steps.**
-
-### Step 0: Prepare Environment
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:11]: Keep step0?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 14]
```diff
@@ -2,26 +2,24 @@
...
-pip install bigdl-orca
-pip install torch==1.7.1 torchvision==0.8.2
-pip install six cloudpickle
-pip install jep==3.9.0
+pip install --pre --upgrade bigdl-orca 
+pip install torch torchvision
+pip install tqdm
 ```
 
+**In this guide we will describe how to scale out _PyTorch_ programs using Orca in 5 simple steps.**
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:21]: put this before step0?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 15]
```diff
@@ -70,44 +63,51 @@ class LeNet(nn.Module):
...
-criterion = nn.NLLLoss()
-adam = torch.optim.Adam(model.parameters(), 0.001)
+def optim_creator(model, config):
+    return torch.optim.Adam(model.parameters(), lr=0.001)
 ```
 
 ### Step 3: Define Train Dataset
 
-You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html). 
+You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Spark Dataframes](../Overview/data-parallel-processing#spark-dataframes) and [XShards](../Overview/data-parallel-processing#xshards-distributed-data-parallel-python-processing).
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: Dataframes -> DataFrames

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 16]
```diff
@@ -70,44 +63,51 @@ class LeNet(nn.Module):
...
 
 ```python
 import torch
 from torchvision import datasets, transforms
 
 torch.manual_seed(0)
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:86]: remove this? as set seed needs to be in the creator

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 17]
```diff
@@ -116,34 +116,32 @@ First, Create an Estimator
...
 Next, fit and evaluate using the Estimator
 
 ```python
-from bigdl.orca.learn.trigger import EveryEpoch 
+est.fit(data=train_loader_creator, epochs=1, batch_size=batch_size)
+```
 
-est.fit(data=train_loader, epochs=10, validation_data=test_loader,
-        checkpoint_trigger=EveryEpoch())
+Finally, evaluate using the Estimator.
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:128]: remove this? as evaluate is already mentioned in `Next, fit and evaluate using the Estimator`

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 18]
```diff
@@ -70,44 +63,51 @@ class LeNet(nn.Module):
         x = F.relu(self.fc1(x))
         x = self.fc2(x)
         return F.log_softmax(x, dim=1)
+```
+After defining your model, you need to define a *Model Creator Function* that returns an instance of your model, and a *Optimizer Creator Function* that returns a PyTorch optimizer.
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:67]: that takes the parameter `config` and returns ...

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 19]
```diff
@@ -2,24 +2,22 @@
 
 ---
 
-![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist.ipynb)
+![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/BigDL/blob/main/python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb)
 
 ---
 
-**In this guide we will describe how to scale out _PyTorch_ programs using Orca in 4 simple steps.**
+**In this guide we will describe how to scale out _PyTorch_ programs using Orca in 6 simple steps.**
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:9]: 6 -> 5

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 20]
```diff
@@ -70,44 +63,50 @@ class LeNet(nn.Module):
...
+    return model
+
+def optim_creator(model, config):
+    return torch.optim.Adam(model.parameters(), lr=0.001)
 ```
 
 ### Step 3: Define Train Dataset
 
-You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html). 
+You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Spark DataFrames](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/data-parallel-processing.html#spark-dataframes) and [XShards](https://bigdl.readthedocs.io/en/latest/doc/Orca/Overview/data-parallel-processing.html#xshards-distributed-data-parallel-python-processing).
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: use the relative path?

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md [hunk 21]
```diff
@@ -70,44 +63,50 @@ class LeNet(nn.Module):
...
+    return model
+
+def optim_creator(model, config):
+    return torch.optim.Adam(model.parameters(), lr=0.001)
 ```
 
 ### Step 3: Define Train Dataset
 
-You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html). 
+You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Spark DataFrames](../Overview/data-parallel-processing.html#spark-dataframes) and [XShards](../Overview/data-parallel-processing.html#xshards-distributed-data-parallel-python-processing).
```
Comment (hkvision) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md:80]: the link of the dataloader is missing

File: python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb (hunk 11 above)
Comment (lalalapotter) [reply python/orca/colab-notebook/quickstart/pytorch_lenet_mnist_spark.ipynb]: fixed

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 19 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md]: fixed.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md (hunk 20 above)
Comment (lalalapotter) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-quickstart.md]: fixed.
